File: Frontend/src/flight/service/FlightService.js
import axios from 'axios';

class FlightService {
    getAllFlights(){
        return axios.get('http://localhost:5010/airline-api/flight/flights');
    }

    addFlight(flight){
        return axios.post('http://localhost:5010/airline-api/flight/createFlight', flight);
    }

    getFlight(flightno){
        return axios.get('http://localhost:5010/airline-api/flight/flights/' + flightno);
    }

    deleteFlight(flightno){
        return axios.delete('http://localhost:5010/airline-api/flight/deleteFlight/' + flightno);
    }
}

export default FlightService;

File: backend/src/main/java/com/cg/flightmgmt/service/FlightService.java
package com.cg.flightmgmt.service;

import java.util.List;

import com.cg.flightmgmt.dto.Flight;
import com.cg.flightmgmt.exception.FlightNotFoundException;

public interface FlightService {
	// create a new flight and add it to the the database
	public Flight addFlight(Flight flight);

	// view a flight using its flight number
	public Flight viewFlight(int flightno) throws FlightNotFoundException;

	// view all flights in the database
	public List<Flight> viewFlight();

	// delete a flight using the flight number
	public void deleteFlight(int flightno);

	// modify a flight using the flight number
	public Flight modifyFlight(Flight flight);
}
